src/python/espressomd/electrostatics.pyx

```diff
@@ -436,6 +445,16 @@ IF P3M == 1:
                 params["tune"] = self._params["tune"]
                 return params
 
+            def tune(self, **tune_params_subset):
+                # update the three necessary parameters if not provided by the
```

Comment:
not the scope of this PR... but do you know why the two P3M interface classes do not inherit from a base class and you have to write the same function twice?

Reply:
to keep it short: no i dont, but i think i also never touched anything in the `P3M` interfaces.